src/wasm/wasm.cpp

```diff
@@ -1001,6 +1001,36 @@ void I31Get::finalize() {
   }
 }
 
+// If an instruction depends on a child for its type, for example call_ref
+// whose type should be the return values of the function signature in its
+// target, then we must handle the case of the child being null. In that case,
+// we can't read signature information from the child, but it doesn't matter, as
+// the call will trap anyhow. We could update the type to be unreachable, but
+// that would violate the invariant that non-branch instructions other than
+// `unreachable` can only be unreachable if they have unreachable children. This
+// makes the result type as close to `unreachable` as possible without
+// actually making it unreachable. TODO: consider just making this
+// unreachable instead (and similar in other GC accessors), although this
+// would currently cause the parser to admit more invalid modules.
+static Type getMaximallyUninhabitable(Type type) {
+  if (type.isRef()) {
+    // We can make a reference uninhabitable.
+    // TODO: Make this exact.
+    return Type(type.getHeapType().getBottom(), NonNullable);
+  } else if (type.isTuple()) {
+    // We can make a tuple's elements uninhabitable.
+    Tuple elems;
+    for (auto t : type) {
+      // TODO: Make this exact.
+      elems.push_back(t.isRef() ? Type(t.getHeapType().getBottom(), NonNullable)
+                                : t);
```

Comment:
This can be simplified if we don't mind one level of recursion.
```suggestion
      elems.push_back(getMaximallyUninhabitable(t));
```

Reply:
I had a similar thought as I moved the code around, but I think it saves little, and might be slower?